src/analysis/vector/geometry_checker/qgsgeometryselfcontactcheck.cpp

```diff
@@ -17,70 +17,67 @@
 #include "qgsgeometryutils.h"
 #include "qgsfeaturepool.h"
 
-void QgsGeometrySelfContactCheck::collectErrors( QList<QgsGeometryCheckError *> &errors, QStringList &/*messages*/, QAtomicInt *progressCounter, const QMap<QString, QgsFeatureIds> &ids ) const
+QList<QgsSingleGeometryCheckError *> QgsGeometrySelfContactCheck::processGeometry( const QgsGeometry &geometry, const QVariantMap &configuration ) const
 {
-  QMap<QString, QgsFeatureIds> featureIds = ids.isEmpty() ? allLayerFeatureIds() : ids;
-  QgsGeometryCheckerUtils::LayerFeatures layerFeatures( mContext->featurePools, featureIds, mCompatibleGeometryTypes, progressCounter, mContext );
-  for ( const QgsGeometryCheckerUtils::LayerFeature &layerFeature : layerFeatures )
+  QList<QgsSingleGeometryCheckError *> errors;
+  const QgsAbstractGeometry *geom = geometry.constGet();
+  for ( int iPart = 0, nParts = geom->partCount(); iPart < nParts; ++iPart )
```

Comment:
Off topic: we really need proper part/ring iterators in QgsGeometry

Reply:
I think if you look at a diff with whitespace removed you won't notice :P

Reply:
Come on - tell me you wouldn't prefer:

     for ( const auto part : geom.parts() )
     {
        for ( const auto ring: part.rings() )
           ...

Get on it! Boy scout that code! ;)